kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2940,6 +2940,14 @@ object KyuubiConf {
       .version("1.8.1")
       .fallbackConf(ENGINE_INITIALIZE_SQL)
 
+  val ENGINE_FLINK_DOAS_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.engine.flink.doAs.enabled")
+      .doc("Whether to enable using hadoop proxy user to run flink engine. Only takes effect" +
+        s" in kerberos environment and when `${ENGINE_DO_AS_ENABLED.key}` is set to `true`.")
```

Comment:
this configuration should be independent with `kyuubi.engine.doAs.enabled`. And we should fail the Flink engine bootstrap when user enable the configuration on non-Kerberized environment.

the docs might be: "When enabled, the session user is used as the proxy user to launch the Flink engine, otherwise, use the server user. Note, due to the limitation of Apache Flink, it can only be enabled when on Kerberized environment."